### ISSUE TITLE
feat: adamw support for weight decay scheudles

### ISSUE THREAD

**Issue by Dhia-naouali** [google-deepmind/optax#1426]
hi all
wanted to ask if it would be welcome to add weight decay schedules support in adamw directly, without `inject_hyperparams` as it's been used in many SOTA models and research
I would suggest using a `base.ScalarOrSchedule` values instead of a plane float just like the learning_rate arg

if so I'd be happy to work on it

```python

# current implementation / alias def
def adamw(
    learning_rate: base.ScalarOrSchedule,
    b1: float = 0.9,
    b2: float = 0.999,
    eps: float = 1e-8,
    eps_root: float = 0.0,
    mu_dtype: Optional[Any] = None,
    weight_decay: float = 1e-4,
    mask: Optional[Union[Any, Callable[[base.Params], Any]]] = None,
    *,
    nesterov: bool = False,
) -> base.GradientTransformationExtraArgs:

```

**Comment by rdyro** [comment]
Thanks for reporting this, with [#1427](https://github.com/google-deepmind/optax/pull/1427) this should hopefully work now.

Can you give it a try and see if this works for you?

**Comment by Dhia-naouali** [comment]
LGTM
